### ISSUE TITLE
lora don't work! OOM

### ISSUE THREAD

**Issue by zhangguoxin1** [huggingface/trl#2780]
### Reproduction

```python
python examples/scripts/nash_md.py \
    --model_name_or_path ../../autodl-tmp/models/qwen2_7B  \
    --reward_model ../../autodl-tmp/models/qwen2_reward_0_5B \
    --dataset_name ../datas/hh-rlhf-helpful-base \
    --max_length 1024 \
    --learning_rate 1.0e-5 \
    --output_dir ../../autodl-tmp/saves/qwen2_7B_nash_KLfree \
    --per_device_train_batch_size 1 \
    --gradient_accumulation_steps 8 \
    --num_train_epochs 1 \
    --max_new_tokens 128 \
    --warmup_ratio 0.1 \
    --use_peft True \
    --lora_task_type "CAUSAL_LM" \
    --lora_dropout 0.1 \
    --lora_r 8 \
    --lora_alpha 16 \
    --lora_target_modules "q_proj,k_proj,v_proj" \
    --mixture_coef 0 \
    --run_name nash \
    --temperature 0.7 \
    --max_steps 375 \
    --report_to wandb
```

### System Info

- Platform: Linux-5.15.0-78-generic-x86_64-with-glibc2.35
- Python version: 3.10.8
- PyTorch version: 2.1.2+cu118
- CUDA device(s): NVIDIA GeForce RTX 4090
- Transformers version: 4.46.1
- Accelerate version: 1.0.1
- Accelerate config: not found
- Datasets version: 3.1.0
- HF Hub version: 0.28.1
- TRL version: 0.15.0.dev0
- bitsandbytes version: not installed
- DeepSpeed version: not installed
- Diffusers version: not installed
- Liger-Kernel version: not installed
- LLM-Blender version: 0.0.2
- OpenAI version: not installed
- PEFT version: 0.12.0

### Checklist

- [x] I have checked that my issue isn't already filed (see [open issues](https://github.com/huggingface/trl/issues?q=is%3Aissue))
- [x] I have included my system information
- [x] Any code provided is minimal, complete, and reproducible ([more on MREs](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks))
- [x] Any code provided is properly formatted in code blocks, (no screenshot, [more on code blocks](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks))
- [x] Any traceback provided is complete

**Comment by MilchstraB** [comment]
I also encountered this problem. I solved it by downgrading the trl version to `0.13.0`.

**Comment by albertvillanova** [comment]
Hey @zhangguoxin1 — thanks for the report.

A couple of things have shifted since you filed this: the Nash-MD trainer was moved into `trl.experimental.nash_md` (so it's no longer part of the stable API), and the example script's flag was renamed from `--reward_model` to `--reward_model_path`, so your command wouldn't even parse on current `main`. On top of that, fitting a 7B policy + 7B-ish reward model + reference model for Nash-MD on a single 24 GB 4090 is going to be very tight regardless of TRL version — that's likely the underlying OOM rather than a bug.

I'm going to close as outdated, but happy to revisit if you can reproduce on a recent TRL release and share the full traceback + `trl env` output.
